EPI/parse-nested-GrFN.py

```diff
@@ -0,0 +1,124 @@
+import json
+from argparse import ArgumentParser
+
+
+def formatGraph(data): 
+    modelMetadata = {}
+    nodesDict = {}
+    nodes = []
+    edges = []
+
+    for key,value in data.items():
+        if key == 'variables':
+            for item in value: 
+                if 'metadata' in item:
+                    metadata = item['metadata'][0]
+                else:
+                    metadata = {}
+                item['nodeType'] = 'variable'
+                item['metadata'] = metadata
+                nodesDict[item['uid']] = item
+        if key == 'functions':
+            for item in value: 
+                if 'metadata' in item:
+                    metadata = item['metadata'][0]
+                else:
+                    metadata = {} 
+                item['nodeType'] = 'function'
+                item['metadata'] = metadata
+                nodesDict[item['uid']] = item
+        if key == 'hyper_edges':
+            for item in value: 
+                if len(item['inputs']) > 0:
+                    for i in item['inputs']: # Inputs array is sometimes empty
+                        first_edge = { 'source': i, 'target': item['function']}
+                        edges.append(first_edge)
+                
+                for o in item['outputs']:
+                    second_edge = { 'source': item['function'], 'target': o }
+                    edges.append(second_edge)
```

Comment:
We can do this whole for-loop in one line:
```
[edges.append({ 'source': item['function'], 'target': o }) for o in item['outputs']]
```
😄

Reply:
Done, thanks!